karapace/backup/consumer.py

```diff
@@ -0,0 +1,45 @@
+"""
+Copyright (c) 2023 Aiven Ltd
+See LICENSE for details
+"""
+from datetime import timedelta
+from isodate import duration_isoformat, parse_duration
```

Comment:
question: [python-dateutil](https://github.com/dateutil/dateutil) is the usual go-to to fill the needs that stdlib datetime doesn't fulfill. Have you checked whether it can be used instead?

Reply:
It does not support ISO8601 durations.